Revert recycle bin location change

The change of the recycle bin location produced an error while executing related migration code


From @dalvizu 
```
sudo docker logs swirlds-node
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> BEGIN USER IDENT >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
uid=2000(hedera) gid=2000(hedera) groups=2000(hedera)
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<< END USER IDENT   <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<

Exception in thread "main" java.io.UncheckedIOException: java.nio.file.AtomicMoveNotSupportedException: data/saved/swirlds-recycle-bin -> data/recycle-bin: Invalid cross-device link
	at com.swirlds.common.io.utility.FileUtils.rethrowIO(FileUtils.java:73)
	at com.swirlds.common.io.filesystem.internal.FileSystemManagerImpl.<init>(FileSystemManagerImpl.java:108)
	at com.swirlds.common.io.filesystem.internal.FileSystemManagerFactoryImpl.createFileSystemManager(FileSystemManagerFactoryImpl.java:58)
	at com.swirlds.common.context.DefaultPlatformContext.<init>(DefaultPlatformContext.java:59)
	at com.swirlds.platform.builder.PlatformBuilder.buildPlatformContext(PlatformBuilder.java:374)
	at com.hedera.node.app.ServicesMain.main(ServicesMain.java:161)
Caused by: java.nio.file.AtomicMoveNotSupportedException: data/saved/swirlds-recycle-bin -> data/recycle-bin: Invalid cross-device link
	at java.base/sun.nio.fs.UnixFileSystem.move(UnixFileSystem.java:871)
	at java.base/sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:309)
	at java.base/java.nio.file.Files.move(Files.java:1430)
	at com.swirlds.common.io.filesystem.internal.FileSystemManagerImpl.lambda$new$4(FileSystemManagerImpl.java:108)
	at com.swirlds.common.io.utility.FileUtils.rethrowIO(FileUtils.java:71)
	... 5 more
```



```
[12:41](https://swirldslabs.slack.com/archives/C06R5C8271B/p1715787664905159)
the swirlds-node container, oddly, is still running. That is the entireity of it’s docker log, and there is no swirlds.log
danalvizu@staging-sm-1:~$ sudo docker ps
CONTAINER ID   IMAGE                                              COMMAND                  CREATED        STATUS                  PORTS     NAMES
ca5a73adebff   local/main-network-node:0.50.0-alpha.1             "/opt/hgcapp/service…"   15 hours ago   Up 15 hours                       swirlds-node
```

--- a/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/context/PlatformContext.java
+++ b/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/context/PlatformContext.java
@@ -24,6 +24,7 @@ import com.swirlds.common.crypto.Cryptography;
 import com.swirlds.common.crypto.CryptographyHolder;
 import com.swirlds.common.io.filesystem.FileSystemManager;
 import com.swirlds.common.io.filesystem.FileSystemManagerFactory;
+import com.swirlds.common.io.utility.NoOpRecycleBin;
 import com.swirlds.common.metrics.noop.NoOpMetrics;
 import com.swirlds.config.api.Configuration;
 import com.swirlds.metrics.api.Metrics;
@@ -42,9 +43,11 @@ public interface PlatformContext {
 
     /**
      * Creates a new instance of the platform context. The instance uses a {@link NoOpMetrics} implementation for
-     * metrics. The instance uses the static {@link CryptographyHolder#get()} call to get the cryptography. The instance
+     * metrics and a {@link com.swirlds.common.io.utility.NoOpRecycleBin}.
+     * The instance uses the static {@link CryptographyHolder#get()} call to get the cryptography. The instance
      * uses the static {@link Time#getCurrent()} call to get the time.
      *
+     * @apiNote This method is meant for utilities and testing and not for a node's production operation
      * @param configuration the configuration
      * @return the platform context
      * @deprecated since we need to remove the static {@link CryptographyHolder#get()} call in future.
@@ -54,11 +57,8 @@ public interface PlatformContext {
     static PlatformContext create(@NonNull final Configuration configuration) {
         final Metrics metrics = new NoOpMetrics();
         final Cryptography cryptography = CryptographyHolder.get();
-        // This new method is used by commands that have no access to nodeId,
-        // we temporary assigning node 0 until we remove recycle bin from the fsManager,
-        // then we can set a dummy impl of the recycle bin
         final FileSystemManager fileSystemManager =
-                FileSystemManagerFactory.getInstance().createFileSystemManager(configuration, metrics);
+                FileSystemManagerFactory.getInstance().createFileSystemManager(configuration, new NoOpRecycleBin());
 
         return create(configuration, metrics, cryptography, fileSystemManager);
     }

--- a/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/context/PlatformContext.java
+++ b/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/context/PlatformContext.java
@@ -23,6 +23,7 @@ import com.swirlds.common.context.internal.PlatformUncaughtExceptionHandler;
 import com.swirlds.common.crypto.Cryptography;
 import com.swirlds.common.crypto.CryptographyHolder;
 import com.swirlds.common.io.filesystem.FileSystemManager;
+import com.swirlds.common.io.filesystem.FileSystemManagerFactory;
 import com.swirlds.common.metrics.noop.NoOpMetrics;
 import com.swirlds.config.api.Configuration;
 import com.swirlds.metrics.api.Metrics;
@@ -53,7 +54,13 @@ public interface PlatformContext {
     static PlatformContext create(@NonNull final Configuration configuration) {
         final Metrics metrics = new NoOpMetrics();
         final Cryptography cryptography = CryptographyHolder.get();
-        return create(configuration, metrics, cryptography);
+        // This new method is used by commands that have no access to nodeId,
+        // we temporary assigning node 0 until we remove recycle bin from the fsManager,
+        // then we can set a dummy impl of the recycle bin
+        final FileSystemManager fileSystemManager =
+                FileSystemManagerFactory.getInstance().createFileSystemManager(configuration, metrics);
+
+        return create(configuration, metrics, cryptography, fileSystemManager);
     }
 
     /**
@@ -61,20 +68,23 @@ public interface PlatformContext {
      * <p>
      * The instance uses the static {@link Time#getCurrent()} call to get the time.
      *
-     * @param configuration the configuration
-     * @param metrics       the metrics
-     * @param cryptography  the cryptography
+     * @param configuration     the configuration
+     * @param metrics           the metrics
+     * @param cryptography      the cryptography
+     * @param fileSystemManager the fileSystemManager
      * @return the platform context
      */
     @NonNull
     static PlatformContext create(
             @NonNull final Configuration configuration,
             @NonNull final Metrics metrics,
-            @NonNull final Cryptography cryptography) {
+            @NonNull final Cryptography cryptography,
+            @NonNull final FileSystemManager fileSystemManager) {
         final Time time = Time.getCurrent();
         final UncaughtExceptionHandler handler = new PlatformUncaughtExceptionHandler();
         final ExecutorFactory executorFactory = ExecutorFactory.create("platform", null, handler);
-        return new DefaultPlatformContext(configuration, metrics, cryptography, time, executorFactory);
+        return new DefaultPlatformContext(
+                configuration, metrics, cryptography, time, executorFactory, fileSystemManager);
     }
 
     /**

--- a/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/context/internal/DefaultPlatformContext.java
+++ b/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/context/internal/DefaultPlatformContext.java
@@ -21,7 +21,6 @@ import com.swirlds.common.concurrent.ExecutorFactory;
 import com.swirlds.common.context.PlatformContext;
 import com.swirlds.common.crypto.Cryptography;
 import com.swirlds.common.io.filesystem.FileSystemManager;
-import com.swirlds.common.io.filesystem.FileSystemManagerFactory;
 import com.swirlds.config.api.Configuration;
 import com.swirlds.metrics.api.Metrics;
 import edu.umd.cs.findbugs.annotations.NonNull;
@@ -45,27 +44,13 @@ public final class DefaultPlatformContext implements PlatformContext {
     /**
      * Constructor.
      *
-     * @param configuration the configuration
-     * @param metrics       the metrics
-     * @param cryptography  the cryptography
-     * @param time          the time
-     * @param executorFactory the executor factory
+     * @param configuration     the configuration
+     * @param metrics           the metrics
+     * @param cryptography      the cryptography
+     * @param time              the time
+     * @param executorFactory   the executor factory
+     * @param fileSystemManager the fileSystemManager
      */
-    public DefaultPlatformContext(
-            @NonNull final Configuration configuration,
-            @NonNull final Metrics metrics,
-            @NonNull final Cryptography cryptography,
-            @NonNull final Time time,
-            @NonNull final ExecutorFactory executorFactory) {
-        this(
-                configuration,
-                metrics,
-                cryptography,
-                time,
-                executorFactory,
-                FileSystemManagerFactory.getInstance().createFileSystemManager(configuration, metrics));
-    }
-
     public DefaultPlatformContext(
             @NonNull final Configuration configuration,
             @NonNull final Metrics metrics,

--- a/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/config/FileSystemManagerConfig.java
+++ b/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/config/FileSystemManagerConfig.java
@@ -40,5 +40,5 @@ public record FileSystemManagerConfig(
 
     public static final String DEFAULT_TMP_DIR_NAME = "tmp";
     public static final String DEFAULT_DATA_DIR_NAME = "saved";
-    public static final String DEFAULT_RECYCLE_BIN_DIR_NAME = "recycle-bin";
+    public static final String DEFAULT_RECYCLE_BIN_DIR_NAME = "saved/swirlds-recycle-bin";
 }

--- a/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/filesystem/FileSystemManagerFactory.java
+++ b/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/filesystem/FileSystemManagerFactory.java
@@ -17,6 +17,7 @@
 package com.swirlds.common.io.filesystem;
 
 import com.swirlds.common.io.filesystem.internal.FileSystemManagerFactoryImpl;
+import com.swirlds.common.platform.NodeId;
 import com.swirlds.config.api.Configuration;
 import com.swirlds.metrics.api.Metrics;
 import edu.umd.cs.findbugs.annotations.NonNull;
@@ -30,12 +31,25 @@ public interface FileSystemManagerFactory {
      * Creates a {@link FileSystemManager} by searching {@code root} path in the {@link Configuration} class under a
      * property name indicated in {@code rootLocationPropertyName}
      *
-     * @param configuration            the configuration instance to retrieve properties from
+     * @param configuration the configuration instance to retrieve properties from
+     * @param metrics       metrics instance of the platform
+     * @return a new instance of {@link FileSystemManager}
+     */
+    @NonNull
+    FileSystemManager createFileSystemManager(@NonNull Configuration configuration, @NonNull Metrics metrics);
+
+    /**
+     * Creates a {@link FileSystemManager} by searching {@code root} path in the {@link Configuration} class under a
+     * property name indicated in {@code rootLocationPropertyName}
+     *
+     * @param configuration the configuration instance to retrieve properties from
+     * @param metrics       metrics instance of the platform
+     * @param nodeId        id of the ode configuring this instance
      * @return a new instance of {@link FileSystemManager}
      */
     @NonNull
     FileSystemManager createFileSystemManager(
-            @NonNull final Configuration configuration, @NonNull final Metrics metrics);
+            @NonNull Configuration configuration, @NonNull Metrics metrics, @NonNull NodeId nodeId);
 
     /**
      * Retrieves the default FileSystemManagerFactory instance

--- a/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/filesystem/FileSystemManagerFactory.java
+++ b/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/filesystem/FileSystemManagerFactory.java
@@ -17,6 +17,7 @@
 package com.swirlds.common.io.filesystem;
 
 import com.swirlds.common.io.filesystem.internal.FileSystemManagerFactoryImpl;
+import com.swirlds.common.io.utility.RecycleBin;
 import com.swirlds.common.platform.NodeId;
 import com.swirlds.config.api.Configuration;
 import com.swirlds.metrics.api.Metrics;
@@ -32,11 +33,11 @@ public interface FileSystemManagerFactory {
      * property name indicated in {@code rootLocationPropertyName}
      *
      * @param configuration the configuration instance to retrieve properties from
-     * @param metrics       metrics instance of the platform
+     * @param recycleBin    the recycleBin instance to use
      * @return a new instance of {@link FileSystemManager}
      */
     @NonNull
-    FileSystemManager createFileSystemManager(@NonNull Configuration configuration, @NonNull Metrics metrics);
+    FileSystemManager createFileSystemManager(@NonNull Configuration configuration, @NonNull RecycleBin recycleBin);
 
     /**
      * Creates a {@link FileSystemManager} by searching {@code root} path in the {@link Configuration} class under a

--- a/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/filesystem/internal/FileSystemManagerFactoryImpl.java
+++ b/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/filesystem/internal/FileSystemManagerFactoryImpl.java
@@ -22,11 +22,14 @@ import com.swirlds.base.time.Time;
 import com.swirlds.common.io.config.FileSystemManagerConfig;
 import com.swirlds.common.io.filesystem.FileSystemManager;
 import com.swirlds.common.io.filesystem.FileSystemManagerFactory;
+import com.swirlds.common.io.utility.NoOpRecycleBin;
 import com.swirlds.common.io.utility.RecycleBinImpl;
+import com.swirlds.common.platform.NodeId;
 import com.swirlds.config.api.Configuration;
 import com.swirlds.metrics.api.Metrics;
 import edu.umd.cs.findbugs.annotations.NonNull;
 import java.io.UncheckedIOException;
+import java.nio.file.Path;
 
 public class FileSystemManagerFactoryImpl implements FileSystemManagerFactory {
 
@@ -40,7 +43,7 @@ public class FileSystemManagerFactoryImpl implements FileSystemManagerFactory {
      * {@code FileSystemManagerConfig} record
      *
      * @param configuration the configuration instance to retrieve properties from
-     * @param metrics metrics instance
+     * @param metrics       metrics instance
      * @return a new instance of {@link FileSystemManager}
      * @throws UncheckedIOException if the dir structure to rootLocation cannot be created
      */
@@ -48,14 +51,33 @@ public class FileSystemManagerFactoryImpl implements FileSystemManagerFactory {
     @Override
     public FileSystemManager createFileSystemManager(
             @NonNull final Configuration configuration, @NonNull final Metrics metrics) {
-
         final FileSystemManagerConfig fsmConfig = configuration.getConfigData(FileSystemManagerConfig.class);
+        return new FileSystemManagerImpl(
+                fsmConfig.rootPath(), fsmConfig.userDataDir(), fsmConfig.tmpDir(), new NoOpRecycleBin());
+    }
 
+    /**
+     * Creates a {@link FileSystemManager} and a {@link com.swirlds.common.io.utility.RecycleBin} by searching {@code root}
+     * path in the {@link Configuration} class using
+     * {@code FileSystemManagerConfig} record
+     *
+     * @param configuration the configuration instance to retrieve properties from
+     * @param metrics       metrics instance
+     * @param selfId the node id for the recycle bin path
+     * @return a new instance of {@link FileSystemManager}
+     * @throws UncheckedIOException if the dir structure to rootLocation cannot be created
+     */
+    @NonNull
+    @Override
+    public FileSystemManager createFileSystemManager(
+            @NonNull final Configuration configuration, @NonNull final Metrics metrics, @NonNull final NodeId selfId) {
+        final FileSystemManagerConfig fsmConfig = configuration.getConfigData(FileSystemManagerConfig.class);
+        final Path recycleBinDir = Path.of(fsmConfig.recycleBinDir()).resolve(selfId.toString());
         return new FileSystemManagerImpl(
                 fsmConfig.rootPath(),
                 fsmConfig.userDataDir(),
                 fsmConfig.tmpDir(),
-                fsmConfig.recycleBinDir(),
+                recycleBinDir.toString(),
                 path -> new RecycleBinImpl(
                         metrics,
                         getStaticThreadManager(),

--- a/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/filesystem/internal/FileSystemManagerFactoryImpl.java
+++ b/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/filesystem/internal/FileSystemManagerFactoryImpl.java
@@ -22,7 +22,7 @@ import com.swirlds.base.time.Time;
 import com.swirlds.common.io.config.FileSystemManagerConfig;
 import com.swirlds.common.io.filesystem.FileSystemManager;
 import com.swirlds.common.io.filesystem.FileSystemManagerFactory;
-import com.swirlds.common.io.utility.NoOpRecycleBin;
+import com.swirlds.common.io.utility.RecycleBin;
 import com.swirlds.common.io.utility.RecycleBinImpl;
 import com.swirlds.common.platform.NodeId;
 import com.swirlds.config.api.Configuration;
@@ -43,17 +43,16 @@ public class FileSystemManagerFactoryImpl implements FileSystemManagerFactory {
      * {@code FileSystemManagerConfig} record
      *
      * @param configuration the configuration instance to retrieve properties from
-     * @param metrics       metrics instance
+     * @param recycleBin    the recycle-bin instance to use in the fileSystemManager
      * @return a new instance of {@link FileSystemManager}
      * @throws UncheckedIOException if the dir structure to rootLocation cannot be created
      */
     @NonNull
     @Override
     public FileSystemManager createFileSystemManager(
-            @NonNull final Configuration configuration, @NonNull final Metrics metrics) {
+            @NonNull final Configuration configuration, @NonNull final RecycleBin recycleBin) {
         final FileSystemManagerConfig fsmConfig = configuration.getConfigData(FileSystemManagerConfig.class);
-        return new FileSystemManagerImpl(
-                fsmConfig.rootPath(), fsmConfig.userDataDir(), fsmConfig.tmpDir(), new NoOpRecycleBin());
+        return new FileSystemManagerImpl(fsmConfig.rootPath(), fsmConfig.userDataDir(), fsmConfig.tmpDir(), recycleBin);
     }
 
     /**

--- a/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/filesystem/internal/FileSystemManagerImpl.java
+++ b/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/filesystem/internal/FileSystemManagerImpl.java
@@ -29,7 +29,6 @@ import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.nio.file.StandardCopyOption;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.function.Function;
 
@@ -62,9 +61,42 @@ public class FileSystemManagerImpl implements FileSystemManager {
     private final Path rootPath;
     private final Path tempPath;
     private final Path savedPath;
-    private final Path recycleBinPath;
     private final RecycleBin bin;
     private final AtomicLong tmpFileNameIndex = new AtomicLong(0);
+    /**
+     * Creates a {@link FileSystemManager} and a {@link com.swirlds.common.io.utility.RecycleBin} by searching {@code root}
+     * path in the {@link Configuration} class using
+     * {@code FileSystemManagerConfig} record
+     *
+     * @param rootLocation      the location to be used as root path. It should not exist.
+     * @param dataDirName       the name of the user data file directory
+     * @param tmpDirName        the name of the tmp file directory
+     * @param recycleBin       for building the recycle bin.
+     * @throws UncheckedIOException if the dir structure to rootLocation cannot be created
+     */
+    FileSystemManagerImpl(
+            @NonNull final String rootLocation,
+            final String dataDirName,
+            final String tmpDirName,
+            @NonNull final RecycleBin recycleBin) {
+        this.rootPath = Path.of(rootLocation).normalize();
+        if (!exists(rootPath)) {
+            rethrowIO(() -> Files.createDirectories(rootPath));
+        }
+
+        this.tempPath = rootPath.resolve(tmpDirName);
+        this.savedPath = rootPath.resolve(dataDirName);
+
+        if (!exists(savedPath)) {
+            rethrowIO(() -> Files.createDirectory(savedPath));
+        }
+        if (exists(tempPath)) {
+            rethrowIO(() -> FileUtils.deleteDirectory(tempPath));
+        }
+        rethrowIO(() -> Files.createDirectory(tempPath));
+
+        this.bin = recycleBin;
+    }
 
     /**
      * Creates a {@link FileSystemManager} and a {@link com.swirlds.common.io.utility.RecycleBin} by searching {@code root}
@@ -80,9 +112,9 @@ public class FileSystemManagerImpl implements FileSystemManager {
      */
     FileSystemManagerImpl(
             @NonNull final String rootLocation,
-            final String dataDirName,
-            final String tmpDirName,
-            final String recycleBinDirName,
+            @NonNull final String dataDirName,
+            @NonNull final String tmpDirName,
+            @NonNull final String recycleBinDirName,
             @NonNull final Function<Path, RecycleBin> binSupplier) {
         this.rootPath = Path.of(rootLocation).normalize();
         if (!exists(rootPath)) {
@@ -91,7 +123,7 @@ public class FileSystemManagerImpl implements FileSystemManager {
 
         this.tempPath = rootPath.resolve(tmpDirName);
         this.savedPath = rootPath.resolve(dataDirName);
-        this.recycleBinPath = rootPath.resolve(recycleBinDirName);
+        final Path recycleBinPath = rootPath.resolve(recycleBinDirName);
 
         if (!exists(savedPath)) {
             rethrowIO(() -> Files.createDirectory(savedPath));
@@ -101,15 +133,8 @@ public class FileSystemManagerImpl implements FileSystemManager {
         }
         rethrowIO(() -> Files.createDirectory(tempPath));
 
-        // FUTURE-WORK: --MIGRATION-- Remove this logic after the fs manager was deployed.
-        // Moves files in the old location of the recycle bin to the new one
-        final Path oldRecyclePath = savedPath.resolve("swirlds-recycle-bin");
-        if (!exists(recycleBinPath) && exists(oldRecyclePath)) {
-            rethrowIO(() -> Files.move(oldRecyclePath, recycleBinPath, StandardCopyOption.ATOMIC_MOVE));
-        }
-
         if (!exists(recycleBinPath)) {
-            rethrowIO(() -> Files.createDirectory(recycleBinPath));
+            rethrowIO(() -> Files.createDirectories(recycleBinPath));
         }
         this.bin = binSupplier.apply(recycleBinPath);
     }

--- a/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/utility/NoOpRecycleBin.java
+++ b/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/utility/NoOpRecycleBin.java
@@ -22,7 +22,6 @@ import java.nio.file.Path;
 
 /**
  * A no-op {@link RecycleBin} implementation.
- *
  */
 public class NoOpRecycleBin implements RecycleBin {
 

--- a/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/utility/NoOpRecycleBin.java
+++ b/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/utility/NoOpRecycleBin.java
@@ -1,0 +1,37 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.io.utility;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.nio.file.Path;
+
+/**
+ * A no-op {@link RecycleBin} implementation.
+ *
+ */
+public class NoOpRecycleBin implements RecycleBin {
+
+    @Override
+    public void recycle(@NonNull Path path) throws IOException {}
+
+    @Override
+    public void start() {}
+
+    @Override
+    public void stop() {}
+}

--- a/platform-sdk/swirlds-common/src/test/java/com/swirlds/common/io/filesystem/internal/FileSystemManagerImplTest.java
+++ b/platform-sdk/swirlds-common/src/test/java/com/swirlds/common/io/filesystem/internal/FileSystemManagerImplTest.java
@@ -58,8 +58,7 @@ public class FileSystemManagerImplTest {
                 getTestRootPath(),
                 FileSystemManagerConfig.DEFAULT_DATA_DIR_NAME,
                 FileSystemManagerConfig.DEFAULT_TMP_DIR_NAME,
-                FileSystemManagerConfig.DEFAULT_RECYCLE_BIN_DIR_NAME,
-                p -> mockRecycleBin);
+                mockRecycleBin);
     }
 
     private String getTestRootPath() {
@@ -86,8 +85,7 @@ public class FileSystemManagerImplTest {
                 largeRootLocation,
                 FileSystemManagerConfig.DEFAULT_DATA_DIR_NAME,
                 FileSystemManagerConfig.DEFAULT_TMP_DIR_NAME,
-                FileSystemManagerConfig.DEFAULT_RECYCLE_BIN_DIR_NAME,
-                p -> mockRecycleBin);
+                mockRecycleBin);
         // then
         assertThat(Path.of(largeRootLocation)).isDirectory().isNotEmptyDirectory();
     }
@@ -110,8 +108,7 @@ public class FileSystemManagerImplTest {
         assertThat(dir)
                 .isNotEmptyDirectory()
                 .isDirectoryContaining("glob:**" + FileSystemManagerConfig.DEFAULT_TMP_DIR_NAME)
-                .isDirectoryContaining("glob:**" + FileSystemManagerConfig.DEFAULT_DATA_DIR_NAME)
-                .isDirectoryContaining("glob:**" + FileSystemManagerConfig.DEFAULT_RECYCLE_BIN_DIR_NAME);
+                .isDirectoryContaining("glob:**" + FileSystemManagerConfig.DEFAULT_DATA_DIR_NAME);
         assertThat(tmpDir).isDirectoryNotContaining("glob:{" + String.join(",", tmpFileNames) + "}");
     }
 

--- a/platform-sdk/swirlds-common/src/testFixtures/java/com/swirlds/common/test/fixtures/platform/TestPlatformContextBuilder.java
+++ b/platform-sdk/swirlds-common/src/testFixtures/java/com/swirlds/common/test/fixtures/platform/TestPlatformContextBuilder.java
@@ -25,7 +25,6 @@ import com.swirlds.common.context.PlatformContext;
 import com.swirlds.common.crypto.Cryptography;
 import com.swirlds.common.crypto.CryptographyHolder;
 import com.swirlds.common.io.filesystem.FileSystemManager;
-import com.swirlds.common.io.filesystem.FileSystemManagerFactory;
 import com.swirlds.common.io.utility.RecycleBin;
 import com.swirlds.common.metrics.noop.NoOpMetrics;
 import com.swirlds.common.test.fixtures.TestFileSystemManager;
@@ -54,7 +53,6 @@ public final class TestPlatformContextBuilder {
     private Metrics metrics;
     private Cryptography cryptography;
     private Time time = Time.getCurrent();
-    private boolean buildFileSystemManagerFromConfig = false;
     private BiFunction<Metrics, Time, FileSystemManager> fileSystemManagerProvider;
 
     private TestPlatformContextBuilder() {}
@@ -115,15 +113,8 @@ public final class TestPlatformContextBuilder {
     }
 
     @NonNull
-    public TestPlatformContextBuilder withFileSystemManagerFromConfig() {
-        this.buildFileSystemManagerFromConfig = true;
-        return this;
-    }
-
-    @NonNull
     public TestPlatformContextBuilder withFileSystemManager(
             @NonNull final BiFunction<Metrics, Time, FileSystemManager> fileSystemManagerProvider) {
-        this.buildFileSystemManagerFromConfig = false;
         this.fileSystemManagerProvider = fileSystemManagerProvider;
         return this;
     }
@@ -131,33 +122,9 @@ public final class TestPlatformContextBuilder {
     @NonNull
     public TestPlatformContextBuilder withTestFileSystemManager(
             @NonNull final Path rootPath, @NonNull final BiFunction<Metrics, Time, RecycleBin> recycleBinProvider) {
-        this.buildFileSystemManagerFromConfig = false;
         this.fileSystemManagerProvider = (m, t) -> {
             TestFileSystemManager fsm = new TestFileSystemManager(rootPath);
             fsm.setBin(recycleBinProvider.apply(m, t));
-            return fsm;
-        };
-        return this;
-    }
-
-    @NonNull
-    public TestPlatformContextBuilder withTestFileSystemManager(@NonNull final Path rootPath) {
-        this.buildFileSystemManagerFromConfig = false;
-        this.fileSystemManagerProvider = (m, t) -> {
-            TestFileSystemManager fsm = new TestFileSystemManager(rootPath);
-            fsm.setBin(TestRecycleBin.getInstance());
-            return fsm;
-        };
-        return this;
-    }
-
-    @NonNull
-    public TestPlatformContextBuilder withTestFileSystemManager(
-            @NonNull final Path rootPath, @NonNull final RecycleBin recycleBin) {
-        this.buildFileSystemManagerFromConfig = false;
-        this.fileSystemManagerProvider = (m, t) -> {
-            TestFileSystemManager fsm = new TestFileSystemManager(rootPath);
-            fsm.setBin(recycleBin);
             return fsm;
         };
         return this;
@@ -180,12 +147,10 @@ public final class TestPlatformContextBuilder {
         }
 
         final FileSystemManager fileSystemManager;
-        if (buildFileSystemManagerFromConfig) {
-            fileSystemManager = FileSystemManagerFactory.getInstance().createFileSystemManager(configuration, metrics);
-        } else if (fileSystemManagerProvider != null) {
+        if (fileSystemManagerProvider != null) {
             fileSystemManager = fileSystemManagerProvider.apply(metrics, time);
         } else {
-            fileSystemManager = getDefaultManager();
+            fileSystemManager = getTestFileSystemManager();
         }
 
         final ExecutorFactory executorFactory = ExecutorFactory.create("test", new UncaughtExceptionHandler() {
@@ -233,7 +198,7 @@ public final class TestPlatformContextBuilder {
         };
     }
 
-    private static TestFileSystemManager getDefaultManager() {
+    private static TestFileSystemManager getTestFileSystemManager() {
         Path defaultRootLocation = rethrowIO(() -> Files.createTempDirectory("testRootDir"));
         TestFileSystemManager fsm = new TestFileSystemManager(defaultRootLocation);
         fsm.setBin(TestRecycleBin.getInstance());

--- a/platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformBuilder.java
+++ b/platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformBuilder.java
@@ -34,6 +34,8 @@ import com.swirlds.common.context.PlatformContext;
 import com.swirlds.common.crypto.Cryptography;
 import com.swirlds.common.crypto.CryptographyFactory;
 import com.swirlds.common.crypto.CryptographyHolder;
+import com.swirlds.common.io.filesystem.FileSystemManager;
+import com.swirlds.common.io.filesystem.FileSystemManagerFactory;
 import com.swirlds.common.merkle.crypto.MerkleCryptoFactory;
 import com.swirlds.common.merkle.crypto.MerkleCryptography;
 import com.swirlds.common.merkle.crypto.MerkleCryptographyFactory;
@@ -388,7 +390,9 @@ public final class PlatformBuilder {
         setupGlobalMetrics(configuration);
         final Metrics metrics = getMetricsProvider().createPlatformMetrics(selfId);
 
-        return PlatformContext.create(configuration, metrics, cryptography);
+        final FileSystemManager fileSystemManager =
+                FileSystemManagerFactory.getInstance().createFileSystemManager(configuration, metrics, selfId);
+        return PlatformContext.create(configuration, metrics, cryptography, fileSystemManager);
     }
 
     /**

